### PR TITLE
Refactor of state_change functionality

### PR DESCRIPTION
**Pull request by Gerson2102**
This PR is addressing the refactor specified on this issue: #11818

### CODE REVIEW

**Comment by Gerson2102** [comment]
I'm not sure if this is what I had to do. I left a comment on the corresponding issue.

**Review by mattsse — CHANGES_REQUESTED** [review]
yep this is the first step, next step is removing the old file and updating the imports of where these functions are used

**Comment by Gerson2102** [comment]
Got it, lmk what you think now.

File: crates/engine/util/src/reorg.rs
Comment (mattsse) [inline crates/engine/util/src/reorg.rs:1]: please undo the extra `;` :)

**Review by mattsse — CHANGES_REQUESTED** [review]
this looks good ty,

but I don't want to include the extra `;` please undo these

**Comment by Gerson2102** [comment]
Got it @mattsse, lmk what u think now mate.

File: crates/engine/util/src/reorg.rs [hunk 1]
```diff
@@ -323,7 +325,7 @@ where
     for tx in candidate_transactions {
         // ensure we still have capacity for this transaction
         if cumulative_gas_used + tx.gas_limit() > reorg_target.gas_limit {
-            continue
+            continue;
```
Comment (mattsse) [inline crates/engine/util/src/reorg.rs:328]:
```suggestion
            continue
```

File: crates/rpc/rpc-eth-api/src/helpers/pending_block.rs [hunk 2]
```diff
@@ -300,7 +301,7 @@ pub trait LoadPendingBlock: EthApiTypes {
                 // them as invalid here which removes all dependent transactions from the iterator
                 // before we can continue
                 best_txs.mark_invalid(&pool_tx);
-                continue
+                continue;
```
Comment (mattsse) [inline crates/rpc/rpc-eth-api/src/helpers/pending_block.rs:304]:
```suggestion
                continue
```

**Review by mattsse — APPROVED** [review]

**Pull request merged**